### PR TITLE
[Bug 1198766] tester

### PR DESCRIPTION
**Pull request by willkg**
This lets trigger rule writers test the trigger rule against recent feedback.

r?

### CODE REVIEW

**Comment by willkg** [comment]
To review:

1. read through the code
2. run the tests (or let travis run them--either is fine)
3. suggest fixes for the mediocre js code

@mythmon Can you look through this sometime this week?

**Comment by mythmon** [comment]
Travis doesn't like this :bomb:.

**Comment by willkg** [comment]
It was a network failure on setup. I kicked it off again.

**Comment by willkg** [comment]
^^^ Everything passed.

File: fjord/suggest/providers/trigger/api_views.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+class TriggerRuleMatchViewAPI(rest_framework.views.APIView):
+    authentication_classes = (CSRFLessSessionAuthentication,)
+
+    @classmethod
+    def as_view(cls, **initkwargs):
+        # Enable CORS
+        view = super(TriggerRuleMatchViewAPI, cls).as_view(**initkwargs)
+        view = cors_enabled('*', methods=['POST'])(view)
+        return view
```
Comment (mythmon) [inline fjord/suggest/providers/trigger/api_views.py]: Why the assign-then-return here?

**Comment by mythmon** [comment]
Oh, I forgot that adding comments to commits instead of the diff here makes things annoying. Sorry about that.

File: fjord/suggest/providers/trigger/api_views.py (hunk 1 above)
Comment (willkg) [inline fjord/suggest/providers/trigger/api_views.py]: Mmmm... I had this return the value at 28, then later spent time fiddling with `csrf_exempt` and another decorator and switched it to this form which was easier to read. I never switched it back.

I'll fix it.

**Comment by willkg** [comment]
I'm totally fine with adding comments to the individual commits. The notifications go to a different email address (yay github!), but it's fine by me.

**Comment by willkg** [comment]
Yes. It's a link to seeing a single day on the front page dashboard.

**Comment by willkg** [comment]
^^^ That redoes the api endpoint to use serializers, addresses your code comments so far and cleans up the JS.

**Comment by willkg** [comment]
^^^ And that switches to `Array.map`. I think that covers everything.

**Comment by willkg** [comment]
^^^ Nixes some dead code. Less to review now!

File: fjord/suggest/providers/trigger/api_views.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+from fjord.base.api_utils import StrictArgumentsMixin
+from fjord.base.utils import (
+    analyzer_required,
+    cors_enabled,
+)
+from fjord.feedback.models import Response as FeedbackResponse
+from fjord.suggest.providers.trigger.models import TriggerRuleMatcher
+
+
+class TriggerRuleTestSerializer(StrictArgumentsMixin, serializers.Serializer):
```
Comment (mythmon) [inline fjord/suggest/providers/trigger/api_views.py]: Ooh, I forgot about `StrictArugmentsMixin`. So cool

File: fjord/suggest/providers/trigger/tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+            'url_exists': None
+        }
+        resp = self.client.post(
+            reverse('triggerrule-match'),
+            content_type='application/json',
+            data=json.dumps(data)
+        )
+        assert resp.status_code == 200
+        # This matches everything because it's an empty rule. We'll
+        # verify the count and assume it's correct.
```
Comment (mythmon) [inline fjord/suggest/providers/trigger/tests/test_api.py]: This comment doesn't match the code anymore.

**Comment by mythmon** [comment]
Ok. I think this is good now. Well, not good, but as good as it is going to be. I don't see any glaring issues, and it looks better than when it started. It might be good to handle the error from the ajax call (where you already have a TODO), but I'm not sure what you would do. Maybe `alert("It broke. Sorry.")`.

r+

File: fjord/suggest/providers/trigger/tests/test_api.py (hunk 3 above)
Comment (willkg) [inline fjord/suggest/providers/trigger/tests/test_api.py]: Ahh... you're right. Oops.

**Comment by willkg** [comment]
^^^ Fixes the comment and the adds a mediocre error "handler".

I'll wait for Travis to ok this, then land it.

Thank you so much!

**Comment by willkg** [comment]
Yay Travis!

**Comment by willkg** [comment]
Side note: Fixing the bundles and static files is covered in another bug. Right now it's a big mess. :(

**Pull request merged**